tests/functional/aws-node-sdk/test/multipleBackend/delete/delete.js

```diff
@@ -26,133 +32,128 @@ describeSkipIfNotMultiple('Multiple backend delete', () => {
+            process.stdout.write('Putting object to mem\n');
+            await s3.send(new PutObjectCommand({ 
+                Bucket: bucket, 
+                Key: memObject, 
+                Body: body,
+                Metadata: { 'scal-location-constraint': memLocation } 
+            }));
+            
+            process.stdout.write('Putting object to file\n');
+            await s3.send(new PutObjectCommand({ 
+                Bucket: bucket, 
+                Key: fileObject, 
+                Body: body,
+                Metadata: { 'scal-location-constraint': fileLocation } 
+            }));
+            
+            process.stdout.write('Putting object to AWS\n');
+            await s3.send(new PutObjectCommand({ 
+                Bucket: bucket, 
+                Key: awsObject, 
+                Body: body,
+                Metadata: { 'scal-location-constraint': awsLocation } 
+            }));
+            
+            process.stdout.write('Putting 0-byte object to AWS\n');
+            await s3.send(new PutObjectCommand({ 
+                Bucket: bucket, 
+                Key: emptyObject,
+                Metadata: { 'scal-location-constraint': awsLocation } 
+            }));
+            
+            process.stdout.write('Putting big object to AWS\n');
+            await s3.send(new PutObjectCommand({ 
+                Bucket: bucket, 
+                Key: bigObject, 
+                Body: bigBody,
+                Metadata: { 'scal-location-constraint': awsLocation } 
+            }));
         });
-        after(() => {
+
+        after(async () => {
+            process.stdout.write('Emptying bucket\n');
+            await bucketUtil.empty(bucket);
             process.stdout.write('Deleting bucket\n');
-            return bucketUtil.deleteOne(bucket)
-            .catch(err => {
-                process.stdout.write(`Error deleting bucket: ${err}\n`);
-                throw err;
-            });
+            await bucketUtil.deleteOne(bucket);
         });
 
-        it('should delete object from mem', done => {
-            s3.deleteObject({ Bucket: bucket, Key: memObject }, err => {
-                assert.strictEqual(err, null,
-                    `Expected success, got error ${JSON.stringify(err)}`);
-                s3.getObject({ Bucket: bucket, Key: memObject }, err => {
-                    assert.strictEqual(err.code, 'NoSuchKey', 'Expected ' +
-                        'error but got success');
-                    done();
-                });
-            });
+        it('should delete object from mem', async () => {
+            await s3.send(new DeleteObjectCommand({ Bucket: bucket, Key: memObject }));
+            
+            try {
+                await s3.send(new GetObjectCommand({ Bucket: bucket, Key: memObject }));
+                throw new Error('Expected NoSuchKey error but got success');
+            } catch (err) {
+                assert.strictEqual(err.code, 'NoSuchKey');
```

Comment:
I think as you throw in the try, that we will enter the assert of the catch and fail the check, then we loose the inital error you thrown, and we'll see an asert failure with "undefined" that is not "NoSuchKey", can be misleading

(same other places)